x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/user/AuthenticateRequest.java

```diff
@@ -6,50 +6,38 @@
  */
 package org.elasticsearch.xpack.core.security.action.user;
 
+import org.elasticsearch.Version;
 import org.elasticsearch.action.ActionRequest;
 import org.elasticsearch.action.ActionRequestValidationException;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 
 import java.io.IOException;
 
-public class AuthenticateRequest extends ActionRequest implements UserRequest {
+public class AuthenticateRequest extends ActionRequest {
 
-    private String username;
+    public static final AuthenticateRequest INSTANCE = new AuthenticateRequest();
 
     public AuthenticateRequest(StreamInput in) throws IOException {
         super(in);
-        username = in.readString();
+        if (in.getVersion().before(Version.V_8_4_0)) {
```

Comment:
Sorry, I sat on this too long and you'll need to change the constants.

Reply:
No worries. I updated them.